cvat-core/src/consensus-conflict.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2023 CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import { SerializedAnnotationConsensusConflictData, SerializedConsensusConflictData } from './server-response-types';
+import { ObjectType } from './enums';
+
+export enum ConsensusConflictType {
+    NO_MATCHING_ITEM = 'no_matching_item',
+    FAILED_ATTRIBUTE_VOTING = 'failed_attribute_voting',
+    NO_MATCHING_ANNOTATION = 'no_matching_annotation',
+    ANNOTATION_TOO_CLOSE = 'annotation_too_close',
+    WRONG_GROUP = 'wrong_group',
+    FAILED_LABEL_VOTING = 'failed_label_voting',
+}
+
+export class AnnotationConflict {
+    #jobID: number;
+    #serverID: number;
+    #type: ObjectType;
+    #shapeType: string | null;
+    #description: string;
+    #conflictType: ConsensusConflictType;
+
+    constructor(initialData: SerializedAnnotationConsensusConflictData) {
+        this.#jobID = initialData.job_id;
+        this.#serverID = initialData.obj_id;
+        this.#type = initialData.type;
+        this.#shapeType = initialData.shape_type;
+        this.#conflictType = initialData.conflict_type as ConsensusConflictType;
+
+        const desc = this.#conflictType.split('_').join(' ');
+        this.#description = desc.charAt(0).toUpperCase() + desc.slice(1);
+    }
+
+    get jobID(): number {
+        return this.#jobID;
+    }
+
+    get serverID(): number {
+        return this.#serverID;
+    }
+
+    get type(): ObjectType {
+        return this.#type;
+    }
+
+    get shapeType(): string | null {
+        return this.#shapeType;
+    }
+
+    get conflictType(): ConsensusConflictType {
+        return this.#conflictType;
+    }
+
+    get description(): string {
+        return this.#description;
+    }
+}
+
+export default class ConsensusConflict {
```

Comment:
I belive we need to re-use QualityConfilct class. Now we have too much of code duplication here. The difference in `QualityConflict` and `ConsensusConflict` is not big. We can try to make `generic Conflict` class to adress the difference in `type` and `severity` fields. Something like [this](https://www.tutorialsteacher.com/typescript/typescript-generic-class)

On the second thought, do we really need two types of conflicts(and two classes)? For what reason? Can we just adapt QualityConflict to maybe AnnotationConflict and use is as in quality and in consensus?

Reply:
1. The conflict types are different for both
2. There are certain params which are different like QualityConflicts have warnings and error type conflict whereas this isn't present in ConsensusConflicts

Reply:
1. The conflict types are different for both
2. There are certain params which are different like QualityConflicts have warnings and error type conflict whereas this isn't present in ConsensusConflicts